gcloud-java-dns/src/main/java/com/google/gcloud/dns/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1259 @@
+    ZoneContainer oldValue = projectContainer.zones().putIfAbsent(
+        completeZone.getName(), zoneContainer);
+    if (oldValue != null) {
+      return Error.ALREADY_EXISTS.response(String.format(
+          "The resource 'entity.managedZone' named '%s' already exists", completeZone.getName()));
+    }
+    ManagedZone result = OptionParsers.extractFields(completeZone, fields);
+    try {
+      return new Response(HTTP_OK, jsonFactory.toString(result));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(
+          String.format("Error when serializing managed zone %s.", result.getName()));
+    }
+  }
+
+  /**
+   * Creates a new change, stores it, and if delayChange > 0, invokes processing in a new thread.
+   */
+  Response createChange(String projectId, String zoneName, Change change, String... fields) {
+    ZoneContainer zoneContainer = findZone(projectId, zoneName);
+    if (zoneContainer == null) {
+      return Error.NOT_FOUND.response(String.format(
+          "The 'parameters.managedZone' resource named %s does not exist.", zoneName));
+    }
+    Response response = checkChange(change, zoneContainer);
+    if (response != null) {
+      return response;
+    }
+    Change completeChange = new Change();
+    if (change.getAdditions() != null) {
+      completeChange.setAdditions(ImmutableList.copyOf(change.getAdditions()));
+    }
+    if (change.getDeletions() != null) {
+      completeChange.setDeletions(ImmutableList.copyOf(change.getDeletions()));
+    }
+    /* We need to set ID for the change. We are working in concurrent environment. We know that the
+    element fell on an index between 0 and maxId, so we will reset all IDs in this range (all of
+    them are valid for the respective objects). */
+    ConcurrentLinkedQueue<Change> changeSequence = zoneContainer.changes();
+    changeSequence.add(completeChange);
+    int maxId = changeSequence.size();
+    int index = 0;
+    for (Change c : changeSequence) {
+      if (index == maxId) {
+        break;
+      }
+      c.setId(String.valueOf(++index));
+    }
+    completeChange.setStatus("pending");
+    completeChange.setStartTime(ISODateTimeFormat.dateTime().withZoneUTC()
+        .print(System.currentTimeMillis()));
+    invokeChange(projectId, zoneName, completeChange.getId());
+    Change result = OptionParsers.extractFields(completeChange, fields);
+    try {
+      return new Response(HTTP_OK, jsonFactory.toString(result));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(
+          String.format("Error when serializing change %s in managed zone %s in project %s.",
+              result.getId(), zoneName, projectId));
+    }
+  }
+
+  /**
+   * Applies change. Uses a new thread which applies the change only if DELAY_CHANGE is > 0.
+   */
+  private Thread invokeChange(final String projectId, final String zoneName,
+      final String changeId) {
+    if (delayChange > 0) {
+      Thread thread = new Thread() {
```

Comment:
@aozarov suggests using [this](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Executors.html#newScheduledThreadPool(int,%20java.util.concurrent.ThreadFactory)) and invoke [this](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleWithFixedDelay(java.lang.Runnable,%20long,%20long,%20java.util.concurrent.TimeUnit)) method instead of spawning a thread to wait and then complete a task: